src/app/InteractionModelEngine.h

```diff
@@ -178,19 +179,24 @@ class InteractionModelEngine : public Messaging::ExchangeDelegate, public Comman
     bool MergeOverlappedAttributePath(ClusterInfo * apAttributePathList, ClusterInfo & aAttributePath);
     bool IsOverlappedAttributePath(ClusterInfo & aAttributePath);
 
+    CHIP_ERROR RegisterCommandHandler(CommandHandlerInterface * handler);
+    CHIP_ERROR DeregisterCommandHandler(CommandHandlerInterface * handler);
+    CommandHandlerInterface * FindCommandHandler(EndpointId endpointId, ClusterId clusterId);
+    void DeregisterCommandHandlers(EndpointId endpointId);
```

Comment:
It seems like it might also be logical to offer a DeregisterCommandHandlers(ClusterId clusterId) interface to deregister all commands for a given cluster.

Reply:
I've just been adding methods that were minimally needed by our implementation so far. Perhaps we add this if we need it later?